sw/host/opentitanlib/src/app/conf.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Schema for configuration files, exact encoding json/xml to be worked out.
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum PinMode {
+    Input,
+    PushPull,
+    OpenDrain,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct PinConfiguration {
+    pub name: String,
+    pub mode: Option<PinMode>,
+    pub level: Option<bool>,
+    pub pullup: Option<bool>,
+    pub alias_of: Option<String>, // Name of a pin defined by the transport
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum UartParity {
+    None,
+    Even,
+    Odd,
+    Mark,
+    Space,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum UartStopBits {
+    Stop1,
+    Stop1_5,
+    Stop2,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct UartConfiguration {
+    pub name: String,
+    pub baudrate: Option<u32>,
+    pub parity: Option<UartParity>,
+    pub stopbits: Option<UartStopBits>,
+    pub alias_of: Option<String>, // Name of a uart defined by the transport
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct SpiConfiguration {
+    pub name: String,
+    pub alias_of: Option<String>, // Name of a spi port defined by the transport
+}
+
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum FlashAddressMode {
+    Mode3b,
+    Mode4b,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+// Defines the way to reach and program flash storage
+pub enum FlashDriver {
+    SpiEeprom {
+        size: u32,
+        erase_block_size: u32,
+        erase_opcode: u8,
+        program_block_size: u32,
+        address_mode: FlashAddressMode,
+        spi_bus: String, // Name of a bus defined by the transport
+        // Possibly add more fields to declare SPI mode/speed
+    },
+    SpiExternalDriver {
+        command: String,
+        spi_bus: String, // Name of a bus defined by the transport
+        // Possibly add more fields to declare SPI mode/speed
+    }
```

Comment:
This comment is more about my style preference than a specific criticism:  I'd prefer to make `SpiEeprom` and `SpiExternalDriver` public structs and then include them into a variant enum:

```
pub struct SpiEeprom { ... }
pub struct SpiExternalDriver { ... }

pub enum FlashDriver {
  SpiEeprom(SpiEeprom),
  SpiExternalDriver(SpiExternalDriver),
}
```
The reason I prefer this approach is so that functions can be written which operate exclusively on the `SpiEeprom` (and other) structs rather than having functions which always must destructure `FlashDriver` and handle (or throw errors) if they don't understand a particular variant.  One can then write a dispatcher function which matches the variants that you care about in `FlashDriver` and throws errors on the others.